framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -1102,64 +526,6 @@ private String addClassesForElement(Element element) {
         return path;
     }
 
-    /**
-     * Adds an explicit receiver type to a JavaParser method declaration.
-     *
-     * @param methodDeclaration declaration to add a receiver to
-     */
-    private static void addExplicitReceiver(MethodDeclaration methodDeclaration) {
-        if (methodDeclaration.getReceiverParameter().isPresent()) {
-            return;
-        }
-
-        if (!methodDeclaration.getParentNode().isPresent()) {
-            return;
-        }
-
-        com.github.javaparser.ast.Node parent = methodDeclaration.getParentNode().get();
-        if (!(parent instanceof TypeDeclaration)) {
-            return;
-        }
-
-        TypeDeclaration<?> parentDecl = (TypeDeclaration<?>) parent;
-        ClassOrInterfaceType receiver = new ClassOrInterfaceType();
-        receiver.setName(parentDecl.getName());
-        if (parentDecl.isClassOrInterfaceDeclaration()) {
-            ClassOrInterfaceDeclaration parentClassDecl =
-                    parentDecl.asClassOrInterfaceDeclaration();
-            if (!parentClassDecl.getTypeParameters().isEmpty()) {
-                NodeList<Type> typeArgs = new NodeList<>();
-                for (TypeParameter typeParam : parentClassDecl.getTypeParameters()) {
-                    ClassOrInterfaceType typeArg = new ClassOrInterfaceType();
-                    typeArg.setName(typeParam.getNameAsString());
-                    typeArgs.add(typeArg);
-                }
-
-                receiver.setTypeArguments(typeArgs);
-            }
-        }
-
-        methodDeclaration.setReceiverParameter(new ReceiverParameter(receiver, "this"));
-    }
-
-    /**
-     * Transfers all annotations for {@code annotatedType} and its nested types to {@code target},
-     * which is the JavaParser node representing the same type. Does nothing if {@code
-     * annotatedType} is null.
-     *
-     * @param annotatedType type to transfer annotations from
-     * @param target type to transfer annotation to, must represent the same type as {@code
-     *     annotatedType}
-     */
-    private static void transferAnnotations(
-            @Nullable AnnotatedTypeMirror annotatedType, Type target) {
-        if (annotatedType == null) {
-            return;
-        }
-
-        target.accept(new AnnotationTransferVisitor(), annotatedType);
-    }
-
     ///
     /// Classes and class names
```

Comment:
The section "Classes and class names" in both files consists of static methods whose bodies are identical.  Rather than duplicating them, move them to a superclass or to ElementUtils.java.